driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryptionCorpusTest.java

```diff
@@ -0,0 +1,292 @@
+        dataKeysCollection.insertOne(bsonDocumentFromPath("corpus-key-aws.json"));
+        dataKeysCollection.insertOne(bsonDocumentFromPath("corpus-key-local.json"));
+
+        // Step 4: Configure our objects
+        Map<String, Map<String, Object>> kmsProviders = new HashMap<String, Map<String, Object>>();
+        Map<String, Object> awsCreds = new HashMap<String, Object>();
+        Map<String, Object> localMasterkey = new HashMap<String, Object>();
+        awsCreds.put("accessKeyId", System.getProperty("org.mongodb.test.awsAccessKeyId"));
+        awsCreds.put("secretAccessKey", System.getProperty("org.mongodb.test.awsSecretAccessKey"));
+
+        byte[] localMasterkeyBytes = Base64.getDecoder().decode("Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBM"
+                + "UN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk");
+        localMasterkey.put("key", localMasterkeyBytes);
+        kmsProviders.put("aws", awsCreds);
+        kmsProviders.put("local", localMasterkey);
+
+        HashMap<String, BsonDocument> schemaMap = new HashMap<String, BsonDocument>();
+        schemaMap.put("db.coll", schemaDocument);
+
+        AutoEncryptionSettings.Builder autoEncryptionSettingsBuilder = AutoEncryptionSettings.builder()
+                .keyVaultNamespace("admin.datakeys")
+                .kmsProviders(kmsProviders);
+
+        if (useLocalSchema) {
+            autoEncryptionSettingsBuilder.schemaMap(schemaMap);
+        }
+
+        clientSettings = getMongoClientSettingsBuilder()
+                .codecRegistry(fromRegistries(
+                        fromCodecs(new UuidCodec(UuidRepresentation.STANDARD)), MongoClientSettings.getDefaultCodecRegistry()))
+                .autoEncryptionSettings(autoEncryptionSettingsBuilder.build())
+                .build();
+        autoEncryptingClient = MongoClients.create(clientSettings);
+
+        ClientEncryptionSettings clientEncryptionSettings = ClientEncryptionSettings.builder().
+                keyVaultMongoClientSettings(MongoClientSettings.builder().build()).
+                kmsProviders(kmsProviders).
+                keyVaultNamespace("admin.datakeys").build();
+        clientEncryption = ClientEncryptions.create(clientEncryptionSettings);
+    }
+
+    @Test
+    public void testCorpus() throws IOException, URISyntaxException {
+
+        // Step 5: Iterate over corpus
+        BsonDocument corpus = bsonDocumentFromPath("corpus.json");
+        BsonDocument corpusCopied = new BsonDocument();
+        for (String field : corpus.keySet()) {
+            if (field.equals("_id") || field.equals("altname_aws") || field.equals("altname_local")) {
+                corpusCopied.append(field, corpus.get(field));
+                continue;
+            }
+
+            BsonDocument fieldDocument = corpus.getDocument(field).clone();
+            String kms = fieldDocument.getString("kms").getValue();
+            // TODO: unused
```

Comment:
@kevinAlbs double checking that this is properly unused and can be remove

Reply:
Yes, `type` is not needed to run the test.